core/lib/resources/presence/index.js

```diff
@@ -33,7 +33,7 @@ Presence.prototype.setup = function() {
   var self = this;
 
   this.manager = new PresenceManager(this.name, this.options.policy, Presence.sentry);
-  this.manager.on('user_online', function(userId, userType, userData) {
+  this.manager.on('user_online', function(userId, userType, userData, data) {
```

Comment:
I suggest only on Presence at the moment. I don't want this to grow out of proportion. Maybe estimate first? If it is too much for a single PR, break into more pieces: Say, fix PresenceStore first, then PresenceManager etc...